deployment/roles/common/tasks/repo.yml

```diff
@@ -1,25 +1,23 @@
 ---
-- name: Create repo directory
-  file: 
-    path: "{{ bridge_path }}"
-    state: directory
-
 - name: Register files for copying
   shell: |
-    git ls-tree -r HEAD --name-only
+    git ls-tree -r HEAD --name-only | sed '/^contracts$/d'
     cd contracts; git ls-tree -r HEAD --name-only | sed -e 's/^/contracts\//'
-  register: rsync_files
+  register: registered_files
   delegate_to: 127.0.0.1
   become: false
   args:
-    chdir: ".."
+    chdir: "{{ lookup('env', 'PWD') }}/.."
+
+- name: Create repo directory structure
+  file: 
+    path: "{{ (bridge_path + '/' + item) }}"
+    state: directory
+    recurse: yes
+  with_items: "{{ registered_files.stdout_lines | map('dirname') | list | unique }}"
 
 - name: Copy the files
-  synchronize:
-    dest: "{{ bridge_path }}"
-    src: ../../../..
-    rsync_opts:
-      - "--include=\"{{ item }}\""
-      - "--exclude='*'"
-      - "-avz"
-  with_items: rsync_files.stdout_lines
+  copy:
```

Comment:
This does not create all the parent directories, that's why the previous task to create structure of directories is necessary.